src/async_fuse/memfs/cache/block.rs

```diff
@@ -0,0 +1,306 @@
+//! Utilities for blocks.
+
+use std::{fmt::Formatter, sync::Arc};
+
+use aligned_utils::bytes::AlignedBytes;
+use clippy_utilities::OverflowArithmetic;
+use nix::sys::uio::IoVec;
+
+use crate::async_fuse::fuse::fuse_reply::{AsIoVec, CouldBeAsIoVecList};
+use crate::async_fuse::fuse::protocol::INum;
+
+/// Page Size
+const PAGE_SIZE: usize = 4096;
+
+/// A common coordinate to locate a block.
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq)]
+pub struct BlockCoordinate(pub INum, pub usize);
+
+/// Make a partial slice of `u8` slice for debugging.
+fn make_partial_u8_slice(src: &[u8], start: usize, len: usize) -> String {
+    let end = start.overflow_add(len).min(src.len());
+    let show_length = end.overflow_sub(start);
```

Comment:
By the way, the `overflow_add` and `overflow_sub` is hard to read.
```
let end = min(src.len(), start + len);
```
It's more easy to read.

Reply:
Just using operatoer `+` will violate the clippy inspection rules.